pkg/collector/corechecks/system/disk/diskv2/disk_windows.go

```diff
@@ -0,0 +1,145 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build windows
+
+package diskv2
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	win "golang.org/x/sys/windows"
+	"regexp"
+	"strings"
+	"unsafe"
+)
+
+func compileRegExp(expr string) (*regexp.Regexp, error) {
+	iExpr := fmt.Sprintf("(?i)%s", expr)
+	return regexp.Compile(iExpr)
+}
+
+func (c *Check) fetchAllDeviceLabelsFromLsblk() error {
+	return nil
+}
+
+func (c *Check) fetchAllDeviceLabelsFromBlkidCache() error {
+	return nil
+}
+
+func (c *Check) fetchAllDeviceLabelsFromBlkid() error {
+	return nil
+}
+
+var mpr = win.NewLazySystemDLL("mpr.dll")
+var procWNetAddConnection2W = mpr.NewProc("WNetAddConnection2W")
+
+type netResource struct {
+	Scope       uint32
+	Type        uint32
+	DisplayType uint32
+	Usage       uint32
+	localName   *uint16
+	remoteName  *uint16
+	comment     *uint16
+	provider    *uint16
+}
+
+// RemotePath constructs the remote path based on the mount type.
+// It converts the Type to uppercase to ensure case-insensitive evaluation.
+func (m mount) RemotePath() (string, string) {
+	if strings.TrimSpace(m.Type) == "" {
+		m.Type = "SMB"
+	}
+	// Convert Type to uppercase for case-insensitive comparison
+	normalizedType := strings.ToUpper(strings.TrimSpace(m.Type))
+	if normalizedType == "NFS" {
+		return normalizedType, fmt.Sprintf(`%s:%s`, m.Host, m.Share)
+	}
```

Comment:
In the Python version the type is just used to decide how to build the remote host name, but I don't think there is any normalization done

Reply:
What do you mean by normalization?

Reply:
I believe what @pgimalac refers with normalization is that in the Go version we are calling `ToUpper` where in the Python version we do not:

```python
if mtype and mtype.lower() == "nfs":
    nr.lpRemoteName = r"{}:{}".format(remote_machine, share)
    self.log.debug("Attempting NFS mount: %s", nr.lpRemoteName)
else:
    nr.lpRemoteName = r"\\{}\{}".format(remote_machine, share).rstrip('\\')
    self.log.debug("Attempting SMB mount: %s", nr.lpRemoteName)
```

Reply:
Looking at the code and the example, it seems that @jose-manuel-almaza is normalizing, instead of using `lower` is using `ToUpper` but ultimately is the same behavior

Reply:
Yes I meant changing the value (and returning it), while it's not used later on
I think the code would be simpler if we just had logic similar to the one linked by Gustavo (so just an if block)